pkg/server/server.go

```diff
@@ -19,57 +22,83 @@ type Service interface {
 	Run(ctx context.Context, server *Server) error
 }
 
+// Handle is a type that pairs a gRPC server to a listener
+type Handle struct {
+	Srv *grpc.Server
+	Lis net.Listener
+}
+
+// NewHandle constructs a gRPC/listener handle
+func NewHandle(srv *grpc.Server, lis net.Listener) *Handle {
+	return &Handle{Srv: srv, Lis: lis}
+}
+
+// Serve calls Serve on the gRPC server with the handle's listener
+func (h *Handle) Serve() error {
+	err := h.Srv.Serve(h.Lis)
+	if err == grpc.ErrServerStopped {
+		return nil
+	}
+	return Error.Wrap(err)
+}
+
+// Close closes the gRPC server gracefully and shuts down the listener.
+func (h *Handle) Close() error {
+	h.Srv.GracefulStop()
+	_ = h.Lis.Close()
```

Comment:
Don't ignore all errors. And I'm not sure whether this is necessary, GracefulStop should already close the listener.